routing/pathfind.go

```diff
@@ -534,82 +550,99 @@ func findPath(tx *bolt.Tx, graph *channeldb.ChannelGraph,
 
 		// If this vertex or edge has been black listed, then we'll skip
 		// exploring this edge.
-		if _, ok := ignoredNodes[v]; ok {
+		if _, ok := ignoredNodes[fromVertex]; ok {
 			return
 		}
 		if _, ok := ignoredEdges[edge.ChannelID]; ok {
 			return
 		}
 
+		nodeInfo := distance[toNode]
+
+		amountToSend := nodeInfo.amountToReceive
+
+		// Compute fee based on the amount that needs to be send to the
+		// next node.
+		var fee lnwire.MilliSatoshi
+		if fromVertex != sourceVertex {
+			fee = computeFee(amountToSend, edge)
+		} else {
+			// Source node has no precedessor to pay fee
```

Comment:
This can be written 
```go
fee := computefee(...)
if from == source {
    fee = 0
}
```

Reply:
Also, I think the comment would be better termed something like "If the from node is not the source of this payment, then it will charge a fee for forwarding the payment on it's outgoing channel."

Reply:
In that case, I'd rather swap it around. Use the zero initialized fee unless not source. Better for performance I guess, because computeFee isn't called? Not that it matters that much here, but performance problems usually start small.

Reply:
yep, that's fine by me 👍

Reply:
Removed else clause